src/main/cpp/RLBotInterface/BallPrediction/BallPrediction.cpp

```diff
@@ -77,33 +77,36 @@ int runBallPrediction()
 	{
 		std::this_thread::sleep_for(std::chrono::milliseconds(16));
 
-		ByteBuffer buf = GameFunctions::UpdateLiveDataPacketFlatbuffer();
+		ByteBuffer renderTickBuffer = GameFunctions::UpdateLiveDataPacketFlatbuffer();
+		ByteBuffer rigidBodyBuffer = GameFunctions::UpdateRigidBodyTickFlatbuffer();
 
-		if (buf.size < 4) {
+		if (renderTickBuffer.size < 4 || rigidBodyBuffer.size < 4) {
 			continue; // Don't attempt to deserialize a packet if it's really small.
 		}
 
-		auto packet = flatbuffers::GetRoot<rlbot::flat::GameTickPacket>(buf.ptr);
+		auto renderTickPacket = flatbuffers::GetRoot<rlbot::flat::GameTickPacket>(renderTickBuffer.ptr);
+		auto rigidBodyPacket = flatbuffers::GetRoot<rlbot::flat::RigidBodyTick>(rigidBodyBuffer.ptr);
 
-		auto ball = packet->ball();
+		auto ball = rigidBodyPacket->ball();
 
 		if (ball != nullptr) {
 
-			auto ballPhys = ball->physics();
+			auto ballPhys = ball->state();
 
 			BallPrediction::BallSlice slice;
 			slice.Location = convertVec(ballPhys->location());
 			slice.Velocity = convertVec(ballPhys->velocity());
 			slice.AngularVelocity = convertVec(ballPhys->angularVelocity());
-			slice.gameSeconds = packet->gameInfo()->secondsElapsed();
-			float gravity = packet->gameInfo()->worldGravityZ();
+			slice.gameSeconds = renderTickPacket->gameInfo()->secondsElapsed();
```

Comment:
My understanding is that gameTime is a long that counts up forever. We should add this to the rigidBodyTick if it is not already.

Reply:
Yes, that was my concern aswell. secondsElapsed is the time from the render tick so it wont be synced with the physics tick. Which we probably want to change. It's not as bad as it seems though, the physics tick runs at 120Hz so we would be looking at a max of 8ms error. 

I do know we have the current tick of the physics tick available in the rigidbody packet. But I am not sure how that value correlates to secondsElapsed. It will break ball prediction for most people if these values don't start counting from the same point in time.

Reply:
The current tick is going to be 120 counts a second, have you tried just dividing it by 120 to see how far from game seconds it is?

Reply:
I have had a difference of anywhere between 1 and 5 seconds, it's nowhere near reliable. Maybe there is a non-interpolated time variable somewhere that gets updated every physics tick?

Reply:
I fear we're at the crossroads between backwards compatibility and progress.